AutoCollection/WebSnippet.ts

```diff
@@ -0,0 +1,293 @@
+                                let headers =  snippetInjectionHelper.getContentEncodingFromHeaders(res);
+                                let writeBufferType = undefined;
+                                if (typeof b === "string") {
+                                    writeBufferType = b;
+                                }
+                                if (headers === null || headers === undefined) {
+                                    if (WebSnippet.ValidateInjection(res, a)) {
+                                        arguments[0] = WebSnippet.InjectWebSnippet(res, a, undefined, writeBufferType);
+                                    }
+                                } else if (headers.length) {
+                                    let encodeType = headers[0];
+                                    arguments[0] = WebSnippet.InjectWebSnippet(res, a, encodeType);
+                                }
+                            }
+                        } catch (err) {
+                            Logging.info("inject snippet error: "+ err);
+                        }
+                        return originalHttpsResponseWrite.apply(res,arguments);
+                    }
+
+                    res.end = function wrap(a: Buffer | string | any, b?:Function | string, c?: Function) {
+                        try {
+                            if (isEnabled && isGetHttpsRequest) {
+                                let headers =  snippetInjectionHelper.getContentEncodingFromHeaders(res);
+                                let endBufferType = undefined;
+                                if (typeof b === "string") {
+                                    endBufferType = b;
+                                }
+                                if (headers === null || headers === undefined) {
+                                    if (WebSnippet.ValidateInjection(res, a)) {
+                                        arguments[0] = WebSnippet.InjectWebSnippet(res, a, undefined, endBufferType);
+                                    }
+                                } else if (headers.length) {
+                                    let encodeType = headers[0];
+                                    arguments[0] = WebSnippet.InjectWebSnippet(res, a, encodeType);
+                                }
+                            }
+                        } catch (err) {
+                            Logging.info("inject snippet error: "+ err);
+                        }
+                        return originalHttpsResponseEnd.apply(res,arguments);
+
+                    }
+                    return originalHttpsRequestListener(req,res);
+                }
+                return originalHttpsServer(options, httpsRequestListener);
+
+            }
+          
+        }
+
+    }
+
+    /**
+     * Validate response and try to inject Web snippet
+     */
+    public static ValidateInjection(response: http.ServerResponse, input: string | Buffer): boolean {
```

Comment:
You already handle singleton using INSTANCE, do you need all these methods as static?